src/Npgsql/MultiplexingDataSource.cs

```diff
@@ -56,14 +58,15 @@ sealed class MultiplexingDataSource : PoolingDataSource
         _connectionLogger = dataSourceConfig.LoggingConfiguration.ConnectionLogger;
         _commandLogger = dataSourceConfig.LoggingConfiguration.CommandLogger;
 
-        // TODO: Think about cleanup for this, e.g. completing the channel at application shutdown and/or
-        // pool clearing
-        _ = Task.Run(MultiplexingWriteLoop, CancellationToken.None)
+        _multiplexWriteLoop = Task.Run(MultiplexingWriteLoop, CancellationToken.None)
             .ContinueWith(t =>
             {
-                // Note that we *must* observe the exception if the task is faulted.
-                _connectionLogger.LogError(t.Exception, "Exception in multiplexing write loop, this is an Npgsql bug, please file an issue.");
-            }, TaskContinuationOptions.OnlyOnFaulted);
+                if (t.IsFaulted)
+                {
+                    // Note that we *must* observe the exception if the task is faulted.
```

Comment:
Is there a reason to change to check IsFaulted inside rather then having OnlyOnFaulted outside?

Plus, suggested comment change:

```suggestion
                    // Note that MultiplexingWriteLoop should never throw an exception - everything should be caught and handled internally.
                    // But we defensively do this just in case, to prevent any unobserved exceptions.
```

Reply:
> Is there a reason to change to check IsFaulted inside rather then having OnlyOnFaulted outside?

Yeah, since that continuation only runs on faulted tasks, in other cases awaiting it will result in throwing `TaskCancelledException`.

Reply:
Good catch